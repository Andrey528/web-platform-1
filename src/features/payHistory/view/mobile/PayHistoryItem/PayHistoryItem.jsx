import React from 'react';
import PropTypes from 'prop-types';
import block from 'bem-cn';
import SVGInline from 'react-svg-inline';

import PaymentSVG from './img/payment.svg';
import WithdrawalSVG from './img/withdrawal.svg';
import './PayHistoryItem.scss';

export const PayHistoryItem = ({ item, locale }) => {
  const b = block('pay-history-item');
  const { date, type, status, amount, purse, paymentMode } = item;
  return (
    <div className={b()}>
      <div className={b('main')}>
        <div className={b('left')}>
          <span className={b('type')}>
            {type === 'payment' ? locale.paymentTypes[status] : locale.withdrawalStatuses[item.status]}
          </span>
          <span className={b('date')}>{date}</span>
          <span className={b('purse')}>{purse ? `${paymentMode}: ${purse}` : ''}</span>
        </div>
        <div className={b('right')}>
          <span className={b('amount')}>
            <span className={b('amount-title')}>{`${locale.amount}: `}</span>
            <span className={b('amount-value')}>{amount}</span>
          </span>
        </div>
      </div>
    </div>
  );
};

PayHistoryItem.propTypes = {
  item: PropTypes.object.isRequired,
  locale: PropTypes.object.isRequired,
};