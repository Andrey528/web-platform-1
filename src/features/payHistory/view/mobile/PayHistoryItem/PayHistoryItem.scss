@import 'shared/style/varDesktop.scss';
.pay-history-item {
  display: flex;
  min-height: 5.5rem;
  border-radius: 0.5rem;
  align-items: center;
  background: #182533;
  box-sizing: border-box;
  margin-bottom: 0.625rem;
  color: white;
  font-weight: 400;
  padding: 1rem;
  &__main {
    flex-grow: 1;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
  }
  &__left {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
  }
  &__right {
    height: 100%;
    position: relative;
    margin-left: auto;
    margin-top: auto;
  }

  &__icon {
    display: block;
    width: 3rem;
  }

  &__type {
    font-size: 1rem;
  }

  &__date {
    font-size: 0.875rem;
  }

  &__purse {
    font-style: italic;
    font-size: 0.9rem;
    color: #999999;
    margin-top: 0.125rem;
  }

  &__amount {
    margin-top: auto;
    display: flex;
    justify-content: flex-end;
  }

  &__amount-title {
    font-size: 0.9rem;
    font-weight: bold;
  }

  &__amount-value {
    font-size: 0.9rem;
    font-weight: bold;
    color: #429BDB;
  }
}