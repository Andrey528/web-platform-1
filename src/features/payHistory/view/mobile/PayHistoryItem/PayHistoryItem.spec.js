import React from 'react';
import { shallow } from 'enzyme';
import block from 'bem-cn';

import { PayHistoryItem } from './PayHistoryItem';

describe('PayHistoryItem tests', () => {
  const b = block('pay-history-item');
  const props = {
    locale: {
      amount: 'amount test',
      withdrawal: 'withdrawal test',
      withdrawalStatuses: { 0: 'Created', 1: 'During', 2: 'Paid out', 3: 'Blocked', 4: 'Canceled' },
      paymentTypes: { 1: 'Refill balance', 6: 'Bet win', 10: 'Withdrawal' },
    },
    item:
    { date: '18.07.2019 21:43', id: 10, paymentMode: 'Card', purse: null, status: 1, amount: 50, type: 'withdrawal' },
  };

  describe('PayHistoryItem with props', () => {
    const payHistoryItem = shallow(<PayHistoryItem {...props} />);
    const { locale } = props;
    const { date, amount } = props.item;

    it('render properly', () => {
      expect(payHistoryItem).toMatchSnapshot();
    });

    describe('render payHistoryItems', () => {
      it('render payHistoryItem type', () => {
        expect(payHistoryItem.find('.pay-history-item__type').text()).toEqual(locale.withdrawalStatuses[1]);
      });

      it('render payHistoryItem date', () => {
        expect(payHistoryItem.find('.pay-history-item__date').text()).toEqual(date);
      });

      it('render payHistoryItem purse', () => {
        expect(payHistoryItem.find('.pay-history-item__purse').text()).toEqual('');
      });

      it('render payHistoryItem amount-title', () => {
        expect(payHistoryItem.find('.pay-history-item__amount-title').text()).toEqual(`${locale.amount}: `);
      });

      it('render payHistoryItem amount-value', () => {
        expect(payHistoryItem.find('.pay-history-item__amount-value').text()).toEqual(amount.toString());
      });
    });

    it('render SVGInline', () => {
      expect(payHistoryItem.find('SVGInline')).toHaveLength(1);
    });

    it('have modificators', () => {
      const className = b('icon').toString();
      expect(payHistoryItem.find('SVGInline').props().className.toString()).toEqual(className.toString());
    });
  });
});