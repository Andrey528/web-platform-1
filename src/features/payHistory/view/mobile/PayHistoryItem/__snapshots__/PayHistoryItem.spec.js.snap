// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayHistoryItem tests PayHistoryItem with props render properly 1`] = `
<div
  className="pay-history-item"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <SVGInline
      className="pay-history-item__icon"
      classSuffix="-svg"
      cleanup={Array []}
      cleanupExceptions={Array []}
      component="span"
      svg="test-file-stub"
    />
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <span
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      During
    </span>
    <span
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      18.07.2019 21:43
    </span>
    <span
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    />
    <span
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <span
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        amount test: 
      </span>
      <span
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        50
      </span>
    </span>
  </div>
</div>
`;
